### ISSUE TITLE
Add lint, vet targets to magefile

### ISSUE THREAD

**Issue by carolynvs** [getporter/porter#2080]
Before we can call ([staticcheck](https://staticcheck.io/docs/getting-started/)) and go vet as part of our build, we need targets for them in our magefile.

1. Checkout the release/v1 branch and submit your PR to that branch when ready.
1. Add EnsureStaticCheck target to our [magefiles](https://github.com/getporter/magefiles) repository, similar to how we install other tools.
2. Update our go mod reference to use this target in Porter.
3. Add a Lint target that calls staticcheck
4. Add a Vet target that calls go vet

Don't worry about fixing problems from the lint and vet targets, that is in another issue.

See our [Contributing Tutorial](https://release-v1.porter.sh/contribute/tutorial) and [New Contributor Guide](https://github.com/getporter/porter/blob/release/v1/CONTRIBUTING.md) for help getting started contributing to Porter.

**Comment by tchaudhry91** [comment]
Hi,
I'm taking a stab at this, I've added the EnsureStaticCheck in the magefiles repo (referenced above). I'll continue further with vet/lint once that is merged.
